app/controllers/api/object_definitions_controller.rb

```diff
@@ -0,0 +1,17 @@
+module Api
+  class ObjectDefinitionsController < BaseController
+    def show
+      object = fetch_object_definition(params['c_id'])
+      render_resource :object_definitions, object
+    end
+
+    private
+
+    def fetch_object_definition(id)
+      klass = collection_class(:object_definitions)
+      obj = klass.find_by(:name => id) || klass.find(id)
+      raise ActiveRecord::RecordNotFound unless obj
```

Comment:
can you show a sample resource ?  can we filter on id, name and other attributes ?